### ISSUE TITLE
⌨ Incorrect heading ranks on manager dashboard

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#11974]
## ⌨ Accessibility Issue

We have two level 2 headings on the manager dashboard that I believe should be nested.

### WCAG 2.1 Criteria

- https://www.w3.org/TR/WCAG20-TECHS/G141.html#:~:text=To%20facilitate%20navigation%20and%20understanding,or%20h4%2C%20etc.).

## 🦋 Expected Behaviour

Headings are nested to properly represent the structure of the document.

## 🕵️ Details

The heading for "Your talent requests" seems to be a subsection underneath the "Your manager tools" heading. However, they are both of rank 2.

## 📸 Screenshot

![Screenshot 2024-11-14 155336](https://github.com/user-attachments/assets/b6f1c334-9c8b-415c-9211-479c90052747)

## 🙋‍♀️ Proposed Solution

Update the "Your talent requests" heading to be an `h3`

## ✅ Acceptance Criteria

- [ ] "Your manager tools" section has proper heading nesting
- [ ] No visual changes
